pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -1120,6 +1127,74 @@ func (r ReconcileHyperConverged) amendBadDefaults(req *common.HcoRequest) (bool,
 	return modified, nil
 }
 
+func (r ReconcileHyperConverged) applyUpgradePatches(req *common.HcoRequest) (bool, error) {
+	modified := false
+
+	knownHcoVersion, _ := req.Instance.Status.GetVersion(hcoVersionName)
+	if knownHcoVersion == "" {
+		knownHcoVersion = "0.0.0"
+	}
+
+	err := readUpgradePatchesFromFile(req)
+	if err != nil {
+		return false, err
+	}
+	err = validateUpgradePatches(req)
+	if err != nil {
+		return false, err
+	}
+
+	knownHcoSV, err := semver.Parse(knownHcoVersion)
+	if err != nil {
+		req.Logger.Error(err, "Error!")
+		return false, err
+	}
```

Comment:
Please move up, just after reading the version. We don't want to do more actions if we know we need to exit anyway.

Reply:
done